### PR TITLE
show conditional element access sig-help after arbitrary expressions

### LINKED ISSUE #32: Code completion for Null conditional element access operator only works on simple name expressions
Open up VS 2015 CTP 5 and type:

String s = "g";
s?[    (At this point the parameter completion popup appears as it's supposed to)

Now try the following expressions:

new String() ?[ (Popup DOESN'T come up)
""?[ (Popup DOESN'T come up)
(" ")?[ (Popup DOESN'T come up)

### PR DESCRIPTION
**Pull request by brettfo**
The old model would only appear after an identifier name.  No change was required for VB but there weren't tests so I added them.  Fixes #32.

Tagging for review: @Pilchie @dpoeschl @jasonmalinowski @rchande @balajikris @basoundr

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
:skull:

File: src/EditorFeatures/CSharp/SignatureHelp/ElementAccessExpressionSignatureHelpProvider.cs [hunk 1]
```diff
@@ -115,7 +116,16 @@ private TextSpan GetTextSpan(ExpressionSyntax expression, SyntaxToken openBracke
         {
             if (openBracket.Parent is BracketedArgumentListSyntax)
             {
-                return CompleteElementAccessExpression.GetTextSpan(expression, openBracket);
+                var conditional = expression.Parent as ConditionalAccessExpressionSyntax;
+                if (conditional != null)
+                {
+                    var start = conditional.Span.Start;
+                    return new TextSpan(start, openBracket.Span.End - start);
```
Comment (jasonmalinowski) [inline src/EditorFeatures/CSharp/SignatureHelp/ElementAccessExpressionSignatureHelpProvider.cs]: There's TextSpan.FromBounds if that's nicer.

File: src/EditorFeatures/CSharp/SignatureHelp/ElementAccessExpressionSignatureHelpProvider.cs [hunk 2]
```diff
@@ -275,7 +302,7 @@ internal static bool TryGetSyntax(SyntaxNode root, int position, ISyntaxFactsSer
         }
 
         /// Error tolerance case for
-        ///     "foo[]"
+        ///     "foo[$$]"
```
Comment (Pilchie) [inline src/EditorFeatures/CSharp/SignatureHelp/ElementAccessExpressionSignatureHelpProvider.cs]: Unintentional?

**Comment by Pilchie** [comment]
:+1:

**Comment by brettfo** [comment]
@basoundr Good catch, I've corrected the TextSpan creation to account for substantia (trivia is anything but trivial).

@jasonmalinowski Thanks for that, TextSpan.FromBounds is exactly what I wanted.

@Pilchie comment change was intentional and I've made it more explicit.

**Pull request merged**